### PR TITLE
Refresh token logic

### PR DESCRIPTION
**Pull request by alyakimenko**
### Description

- Implement refresh token logic
- Fix integration test

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-snowflake/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -41,22 +43,25 @@ type HTTPClient struct {
 }
 
 func New(ctx context.Context, databaseEngine, dbName string) *HTTPClient {
+	client := &HTTPClient{
+		databaseEngine: databaseEngine,
+		dbName:         dbName,
+	}
+
 	retryClient := retryablehttp.NewClient()
 	retryClient.RetryMax = 3
```
Comment (maksenius) [inline client/client.go:52]: lets move it to const

File: client/client.go [hunk 2]
```diff
@@ -120,14 +125,83 @@ func (h *HTTPClient) RunQuery(ctx context.Context, query string) ([]byte, error)
...
+	}
+
+	req.Header.Set("Content-Type", "application/json;charset=UTF-8")
+
+	resp, err := h.httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("do refresh token request: %w", err)
+	}
+
+	if resp.StatusCode != http.StatusOK {
```
Comment (maksenius) [inline client/client.go:160]: Did you check ,  is firebolt return 200?

File: client/client.go (hunk 1 above)
Comment (alyakimenko) [reply client/client.go:52]: Moved

File: client/client.go (hunk 2 above)
Comment (alyakimenko) [reply client/client.go:160]: Yes, Firebolt returns `200` on success

**Pull request merged**